### PR TITLE
Support bin zip files in addition to plugin bin dir

### PR DESCRIPTION
**Pull request by vlaaad**
We now extract `${ext}/plugins/bin/${platform}.zip` files to `/build/plugins/${ext}/plugins/bin/${platform}` folder so that binaries that also need a lot of code files to run can be distributed without slowing down the loading time and without polluting Search in Files/Open Assets dialogs.

Related to #3885

### CODE REVIEW

**Comment by vlaaad** [comment]
Note: I'll make the equivalent bob changes in a separate PR.

File: editor/src/clj/editor/workspace.clj [hunk 1]
```diff
@@ -446,19 +452,40 @@ ordinary paths."
...
+        [Platform/X86_64Linux Platform/X86_64MacOS Platform/X86_64Win32]))
+
+(defn- bin-zip? [resource]
+  (let [path (resource/proj-path resource)]
+    (boolean (some #(string/ends-with? path %) bin-zip-names))))
+
+(defn- unpack-bin-zip! [workspace resource]
+  (let [proj-path (resource/proj-path resource)
+        plugin-file (plugin-path workspace proj-path)
+        infix-path (subs proj-path 0 (- (count proj-path) 4))]
```
Comment (matgis) [inline editor/src/clj/editor/workspace.clj:469]: Might want to assert that `proj-path` ends in `".zip"` here?

**Review by matgis — APPROVED** [review]
Looks good, just a few minor niggles.

File: editor/src/clj/editor/workspace.clj [hunk 2]
```diff
@@ -389,35 +391,39 @@ ordinary paths."
...
-    (when (string/includes? (resource/proj-path resource) "/plugins/bin/")
-      (.setExecutable target-path true))))
+(defn unpack-resource!
+  ([workspace resource]
+   (unpack-resource! workspace nil resource))
+  ([workspace infix-path resource]
+   (let [resource-path (str infix-path (resource/proj-path resource))
+         target-path (plugin-path workspace resource-path)
+         parent-dir (.getParentFile ^File target-path)
+         input-stream (io/input-stream resource)]
```
[the remark this answers is not in the document]
Comment (vlaaad) [reply editor/src/clj/editor/workspace.clj]: Yes, we should!

File: editor/src/clj/editor/workspace.clj [hunk 3]
```diff
@@ -389,35 +391,39 @@ ordinary paths."
...
+     (when-not (.exists parent-dir)
+       (.mkdirs parent-dir))
```
[the remark this answers is not in the document]
Comment (vlaaad) [reply editor/src/clj/editor/workspace.clj]: Good catch!

File: editor/src/clj/editor/workspace.clj (hunk 1 above)
Comment (vlaaad) [reply editor/src/clj/editor/workspace.clj:468]: Yep, fixed

**Review by matgis — APPROVED** [review]

**Pull request merged**